drivers/accel/adxl314/adxl314.c

```diff
@@ -0,0 +1,1146 @@
+	uint8_t reg_value = 0;
+
+	dev = (struct adxl314_dev *)calloc(1, sizeof(*dev));
+
+	if (!dev)
+		return -ENOMEM;
+
+	dev->comm_type = init_param.comm_type;
+
+	if (dev->comm_type == ADXL314_SPI_COMM) {
+		ret = no_os_spi_init(&dev->com_desc.spi_desc, &(init_param.comm_init.spi_init));
+		if (ret)
+			goto error_dev;
+	} else {
+		ret = no_os_i2c_init(&dev->com_desc.i2c_desc, &init_param.comm_init.i2c_init);
+		if (ret)
+			goto error_dev;
+	}
+
+	/* Check for device ID. */
+	ret = adxl314_read_device_data(dev, ADXL314_DEVID_AD, 1, &reg_value);
+	if (ret || (reg_value != ADXL314_DEVID))
+		goto error_com;
+
+	*device = dev;
+
+	/* Set op. mode to Standby. */
+	ret = adxl314_set_op_mode(dev, ADXL314_STDBY);
+	if (ret)
+		goto error_com;
+
+	/* Get Low Power mode status. */
+	ret = adxl314_read_device_data(dev, ADXL314_REG_BW_RATE, 1, &reg_value);
+	if (ret)
+		goto error_com;
+
+	if (reg_value & ADXL314_LOW_POWER_OP)
+		dev->lp_mode = ADXL314_LP_MODE_ON;
+	else
+		dev->lp_mode = ADXL314_LP_MODE_NONE;
+
+	/* Get ODR. */
+	ret = adxl314_read_device_data(dev, ADXL314_REG_BW_RATE, 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->odr = reg_value & ADXL314_RATE_MSK;
+
+	/* Get offsets. */
+	ret = adxl314_read_device_data(dev, ADXL314_REG_OFS_AXIS(0), 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->x_offset = reg_value;
+	ret = adxl314_read_device_data(dev, ADXL314_REG_OFS_AXIS(1), 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->y_offset = reg_value;
+	ret = adxl314_read_device_data(dev, ADXL314_REG_OFS_AXIS(2), 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->z_offset = reg_value;
+
+	/* Get FIFO mode setting. */
+	ret = adxl314_read_device_data(dev, ADXL314_REG_FIFO_CTL, 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->fifo_mode = (reg_value & ADXL314_REG_FIFO_CTL_MODE_MSK) >> 6;
+
+	/* Get FIFO samples setting. */
+	ret = adxl314_read_device_data(dev, ADXL314_REG_FIFO_CTL, 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->fifo_samples = (reg_value & ADXL314_REG_FIFO_CTL_SAMPLES_MSK);
+
+	/* Get dev. threshold for detecting activity. */
+	ret = adxl314_read_device_data(dev, ADXL314_REG_THRESH_ACT, 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->act_thr = reg_value;
+
+	/* Get dev. threshold for detecting inactivity. */
+	ret = adxl314_read_device_data(dev, ADXL314_REG_THRESH_INACT, 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->inact_thr = reg_value;
+
+	/* Get dev. time threshold for detecting inactivity. */
+	ret = adxl314_read_device_data(dev, ADXL314_REG_TIME_INACT, 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->time_inact = reg_value;
+
+	/* Get dev. ACT_INACT_CTL register values. */
+	ret = adxl314_read_device_data(dev, ADXL314_REG_ACT_INACT_CTL, 1, &reg_value);
+	if (ret)
+		goto error_com;
+	dev->act_inact_ctl.value = reg_value;
+
+	/* Put device in Measure mode. */
+	ret = adxl314_set_op_mode(dev, ADXL314_MEAS);
+	if (ret)
+		goto error_com;
+
+	pr_info("ADXL314 successfully initialized.\n");
```

Comment:
I might be wrong, but I don't think there is a point in using  these `pr_info` messages, since we use error codes.

Reply:
The only ones that are left are concerned with the success of the self-test.